### ISSUE TITLE
VSTestV2 not including attachments

### ISSUE THREAD

**Issue by cbrianball** [Microsoft/azure-pipelines-tasks#8625]
I have tried running selenium tests on hosted machines as well as a private machine that I host. I have been unable to see any attachments (screen shots from the web driver) that are added via the unit tests.

When I run the tests through the build pipeline  on a private machine I host, I can view the resulting trx file and verify that it has `ResultFile` elements and that those elements do point to actual image files (I can also zip up the TestResults directory and bring it to my dev machine, unzip it, and open the trx file in Visual studio and I can see the data plus view the images.

I've even been able to view the trx file from Azure DevOps via a URL that is output from the build process. That screen allows me to download the trx file and that file still has the `ResultFile` elements.

Below is the debug output of one of my attempts (running on a privately hosted Windows 10 machine) to get the attachments to be viewable from within Azure DevOps:

I've attached the debut output of the task since it is too long to include inline.
[biuldoutput.txt](https://github.com/Microsoft/azure-pipelines-tasks/files/2484240/biuldoutput.txt)

**Comment by SadagopanRajaram** [comment]
If possible, can you please share the zipped trx folder? You can also open a ticket on https://developercommunity.visualstudio.com

**Comment by cbrianball** [comment]
The only attachment I see in Azure DevOps is the trx file itself, but on the build server (which I host) I can see the files that I've attached.

In the "Visual Studio Test" task, I have the box checked for "Upload test attachments". I'll post another comment next week when I can get to the server and zip the trx folder for you.

**Comment by cbrianball** [comment]
Here's the TestResults folder from the latest execution.
[TestResults.zip](https://github.com/Microsoft/azure-pipelines-tasks/files/2644761/TestResults.zip)

**Comment by SadagopanRajaram** [comment]
Thanks for the zip file. I was able to publish the results and see the attachments in the details for the result iteration. For the failed tests, all the iterations are reported in a hierarchical manner and the inner result details are shown at that lower level in the UI. Let me know if the below documentation helps
https://docs.microsoft.com/en-us/azure/devops/pipelines/test/review-continuous-test-results-after-build?view=vsts#tests-tab

**Comment by cbrianball** [comment]
@SadagopanRajaram How did you publish the results? I do know what you are talking about when you say "inner result", I don't see attachments for any results in Azure DevOps (but if I simply open the trx file from the zip I posted in Visual Studio it works fine.

I am not using any extra tasks to publish the results, I'm relying on the "Visual Studio Test" task to do that -- should I create a separate step to publish the results?

**Comment by SadagopanRajaram** [comment]
This was fixed in Agent 2.142.0 with the below commit. Please update the agent to a version later than 2.140.2 and this should be resolved. 
https://github.com/Microsoft/azure-pipelines-agent/commit/9a5098d1c6d14efde3d7e824e6ea9f0530c30588#diff-a2aa3eddcb5c7bd5be5f375b6de1cea7

**Comment by cbrianball** [comment]
@SadagopanRajaram 
I upgraded to 2.144.0 and subsequent runs of the tests now upload the images in Azure DevOps!

Feedback on upgrade process: I run the agent interactively, so I understand that fully automated updates aren't supported, but I didn't find any information on how to manually perform the update. I found the option in Azure DevOps to push the update, which in my case merely downloaded the files (I found new folders called bin.xxx and externals.xxx where xxx is the new version number). I backed up the existing bin and externals folders and renamed the versioned ones to 'bin' and 'externals', and it looks like that completed the upgrade. Any potential issues with how I did the upgrade?
